test/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.test.cpp

```diff
@@ -1940,11 +1738,1119 @@ TEST_P(OpenSpaceToolkit_Astrodynamics_Trajectory_Segment_Solve_Parameterized, Ma
+    EXPECT_TRUE(solution.states.getSize() > 0);
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_Segment, Solve_MinimumDurationNotDefined)
+{
+    const Segment::ManeuverConstraints constraints(
+        Duration::Undefined(),  // Minimum duration not defined
+        Duration::Undefined(),
+        Duration::Undefined(),
+        Segment::MaximumManeuverDurationViolationStrategy::Fail
+    );
+
+    const Shared<RealCondition> durationCondition = std::make_shared<RealCondition>(
+        RealCondition::DurationCondition(RealCondition::Criterion::AnyCrossing, Duration::Minutes(4.0))
+    );
+
+    const Segment maneuverSegment = Segment::Maneuver(
+        defaultName_,
+        durationCondition,
+        defaultThrusterDynamicsSPtr_,
+        defaultDynamics_,
+        defaultNumericalSolver_,
+        constraints
+    );
+
+    const Segment::Solution solution = maneuverSegment.solve(initialStateWithMass_, Duration::Minutes(5.0));
+
+    EXPECT_TRUE(solution.states.getSize() > 0);
+    EXPECT_GT(solution.extractManeuvers(defaultFrameSPtr_).getSize(), 0);
+    EXPECT_TRUE(solution.conditionIsSatisfied);
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_Segment, Solve_ManeuverDurationMeetsMinimum)
+{
+    const Segment::ManeuverConstraints constraints(
+        Duration::Minutes(1.0),  // Minimum duration defined
+        Duration::Undefined(),
+        Duration::Undefined(),
+        Segment::MaximumManeuverDurationViolationStrategy::Fail
+    );
+
+    const Shared<RealCondition> durationCondition = std::make_shared<RealCondition>(
+        RealCondition::DurationCondition(RealCondition::Criterion::AnyCrossing, Duration::Minutes(2.0))
+    );
+
+    const Segment maneuverSegment = Segment::Maneuver(
+        defaultName_,
+        durationCondition,
+        defaultThrusterDynamicsSPtr_,
+        defaultDynamics_,
+        defaultNumericalSolver_,
+        constraints
+    );
+
+    const Segment::Solution solution = maneuverSegment.solve(initialStateWithMass_, Duration::Minutes(5.0));
+
+    const Array<Maneuver> maneuvers = solution.extractManeuvers(defaultFrameSPtr_);
+    EXPECT_GT(maneuvers.getSize(), 0);
+    EXPECT_TRUE(initialStateWithMass_.accessInstant().isNear(maneuvers[0].getInterval().getStart(), 1e-3));
+}
+
+TEST_F(
+    OpenSpaceToolkit_Astrodynamics_Trajectory_Segment, Solve_ManeuverDurationBelowMinimum_ConditionSatisfiedDuringCoast
+)
+{
+    const Segment::ManeuverConstraints constraints(
+        Duration::Minutes(10.0),  // Minimum duration (large, so short maneuvers will be rejected)
+        Duration::Undefined(),
+        Duration::Undefined(),
+        Segment::MaximumManeuverDurationViolationStrategy::Fail
+    );
+
+    // Event condition that will be satisfied during coast
+    const Shared<InstantCondition> eventCondition = std::make_shared<InstantCondition>(
+        InstantCondition::Criterion::AnyCrossing, initialStateWithMass_.accessInstant() + Duration::Minutes(5.0)
```

Comment:
Why not use a duration condition like the test above and below?
Also, are you sure that it is coasting when the event condition is met? The defaultThrusterDynamicsSPtr_ looks like they are constantly thrusting

```suggestion
    // Event condition that will be satisfied before the Minimum maneuver duration is reached
    const Shared<RealCondition> durationCondition = std::make_shared<RealCondition>(
        RealCondition::DurationCondition(RealCondition::Criterion::AnyCrossing, Duration::Minutes(5.0))
    );
```

Reply:
I can't remember why I started out with Instant Conditions, but they can all be durations, I'll align. 
As for the coasting, it's going to be coasting as it's going to skip the maneuver due to the minimum maneuver duration.